patches/tModLoader/Terraria/ModLoader/ModBackgroundStyle.cs

```diff
@@ -43,44 +48,87 @@ protected override sealed void Register()
 	/// <summary>
 	/// Allows you to modify the transparency of all background styles that exist. In general, you should move the index equal to this style's slot closer to 1, and all other indexes closer to 0. The transitionSpeed parameter is what you should add/subtract to each element of the fades parameter. See the ExampleMod for an example.
 	/// </summary>
-	public abstract void ModifyFarFades(float[] fades, float transitionSpeed);
+	public abstract void ModifyStyleFade(float[] fades, float transitionSpeed);
 
```

Comment:
The original overloads of `ChooseXTexture` and `ModifyFarFades` must be kept present in form of `[Obsolete(error: true)]` methods, with `ModifyFarFades` converted to `virtual`. Refer to 1fc38455f564a9ac462219d467c1ae013002dfb9 and [this PR](https://github.com/tModLoader/tModLoader/commit/1fc38455f564a9ac462219d467c1ae013002dfb9) as a reference. The short summary of "the trick" here is that compile-time erroring methods can only be called from other obsolete methods, so a special non-erroring calling method must be made.

Reply:
I do think Fox has fixed this fully now, open this request changes if you don't think so.